cart/application/cartService.go

```diff
@@ -298,7 +298,7 @@ func (cs *CartService) UpdateItemQty(ctx context.Context, session *web.Session,
 		return err
 	}
 
-	cs.eventPublisher.PublishChangedQtyInCartEvent(ctx, item, qtyBefore, qty, cart.ID)
+	cs.eventPublisher.PublishChangedQtyInCartEvent(ctx, cart, item, qtyBefore, qty)
```

Comment:
Since the event is published before the `behaviourUpdateItem` call the cart will not be updated yet.

Reply:
You are right, updated implementation